src/main.cc

```diff
@@ -1,6 +1,7 @@
 #include <algorithm>
```

Comment:
No signed off in the commit, incorrect title, check [1], it should be : "main: write counters on service stopping"
You should elaborate that signal handlers were added in the description as this is a new "feature".

[1] https://github.com/The-Good-Penguin/tgp-krill-kounter/blob/main/CodingStyle.md

Your branch should be named "ngalea/39-write-counters-on-stop" or similar, check closed PRs how we do it, perhaps this should be part of the coding style ie.  username/#-short-description, created issue for this: https://github.com/The-Good-Penguin/tgp-krill-kounter/issues/95 

NB no need to rename the branch now.